src/app/clusters/time-synchronization-server/time-synchronization-cluster.cpp

```diff
@@ -381,33 +510,25 @@ void TimeSynchronizationServer::OnTimeSyncCompletionFn(TimeSourceEnum timeSource
         }
         return;
     }
-    mGranularity  = granularity;
-    Status status = TimeSource::Set(kRootEndpointId, timeSource);
-    if (!(status == Status::Success || status == Status::UnsupportedAttribute))
-    {
-        ChipLogError(Zcl, "Writing TimeSource failed.");
-    }
+    mGranularity = granularity;
+    mTimeSource  = timeSource;
 }
 
-void TimeSynchronizationServer::OnFallbackNTPCompletionFn(bool timeSyncSuccessful)
+void TimeSynchronizationCluster::OnFallbackNTPCompletionFn(bool timeSyncSuccessful)
 {
     if (timeSyncSuccessful)
     {
         mGranularity = GranularityEnum::kMillisecondsGranularity;
         // Non-matter SNTP because we know it's external and there's only one source
-        Status status = TimeSource::Set(kRootEndpointId, TimeSourceEnum::kNonMatterSNTP);
-        if (!(status == Status::Success || status == Status::UnsupportedAttribute))
-        {
-            ChipLogError(Zcl, "Writing TimeSource failed.");
-        }
+        mTimeSource = TimeSourceEnum::kNonMatterSNTP;
```

Comment:
same here: any changes to attributes need to notify changed. Ember used to do that automatically inside `::Set` members. Please check whenever you replaced a `::Set` and make sure you NotifyAttributeChanged in those cases (and do so only if value changed)